core/src/test/scala/integration/kafka/api/PlaintextAdminIntegrationTest.scala

```diff
@@ -1543,6 +1547,76 @@ class PlaintextAdminIntegrationTest extends BaseAdminIntegrationTest {
     assertNull(returnedOffsets.get(topicPartition))
   }
 
+  @ParameterizedTest(name = TestInfoUtils.TestWithParameterizedQuorumAndGroupProtocolNames)
+  @MethodSource(Array("getTestQuorumAndGroupProtocolParametersClassicGroupProtocolOnly"))
+  def testDeleteRecordsAfterCorruptRecords(quorum: String, groupProtocol: String): Unit = {
```

Comment:
Should this test go in `DeleteRecordsRequestTest`?

Reply:
Sounds good, thanks for the suggestion.

Reply:
I realized that DeleteRecordsRequestTest focuses on testing `DeleteRecordsRequest`, so it might be better to place testDeleteRecordsAfterCorruptRecords to PlaintextAdminIntegrationTest, as it uses admin#deleteRecords without directly invoking DeleteRecordsRequest. WDYT, @divijvaidya?

Reply:
I am a bit worried about `PlaintextAdminIntegrationTest.scala` becoming a very large class with varied different kinds of tests. Having said that, many other DeleteRecords tests reside here currently, so it is ok to add this one.

The refactoring of `PlaintextAdminIntegrationTest` can be left for some other time.